types.cc

```diff
@@ -2023,6 +2023,9 @@ data_value deserialize_aux(const tuple_type_impl& t, View v) {
 
 template<FragmentedView View>
 utils::multiprecision_int deserialize_value(const varint_type_impl&, View v) {
+    while (v.current_fragment().empty()) {
+        v.remove_current();
+    }
```

Comment:
@denesb There's already a `skip_empty_fragments` which does what we want, let's use it.
https://github.com/scylladb/scylladb/blob/14cdd034eea65fffd8875d7a283f1ddcd0e83496/utils/fragment_range.hh#L269-L274

Reply:
But if we really want to be defensive, we also need add a check that the size of the view is greater than 0. Otherwise the subsequent `v.current_fragment().front()` will read garbage again, if v is empty.

Reply:
`skip_empty_fragments` is not part of the `FragmentedView` interface.

Reply:
We can make it be part of it, but I think this change is best kept minimal to facilitate backporting.

Reply:
> skip_empty_fragments is not part of the FragmentedView interface.

I'm not sure what you mean by this. The function takes `FragmentedView` as an argument, and only calls methods from the interface.

Reply:
But whatever, v3 of the PR is good enough anyway.